indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
@gianm another issue is that hadoop libs, sometimes if used by specifying specific coordinates, are only available in special classloader but not added to the druid process classpath (if hadoop libs were in overlord classpath then existing code would work fine and I tested it that way)